### DISCUSSION TITLE
Nextcloud AIO without access to the docker.sock

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#1494] [Ideas] [answered]
Many server hypervisor operating systems do not have the possibility to access the docker.sock or only through complicated detours where you can not be sure if the workaround still works with the next update (for example TrueNAS Scale, Synology, ...). The option to run Nextcloud AIO without the docker.sock would give the possibility to use the advantages of Nextcloud AIO on these systems. And from a cybersecurity point of view, the whole thing would also have an advantage.

Examples of such an implementation:
[https://hub.docker.com/r/iredmail/mariadb](https://hub.docker.com/r/iredmail/mariadb)

**Comment by szaimen** [comment]
Duplicate of https://github.com/nextcloud/all-in-one/discussions/1286

**Reply by ghost** [reply]
unfortunately this does not work and loses the advantages of Nextcloud AIO

**Reply by szaimen** [reply]
Yes however what you are requesting is basically a "fat" container (with all features included) which isn't something we can do as this would mean a complete rewrite which would most likely not even be technically possible and would make maintenance horrorful.

However you might look into the https://hub.docker.com/r/ownyourbits/nextcloudpi/ image which doesn't have all of AIOs feature but at least is a "fat" image.

**Comment by szaimen** [comment]
Hey, I am currently working on a helm-chart that should allow to run the AIO containers on trunas scale: https://github.com/nextcloud/all-in-one/discussions/1582

**Reply by szaimen** [reply]
Btw, it is indeed possible to run AIO on Synology so only platforms that cannot run docker directly like trunas scale are not supported currently but this will change with the above.

**Reply by ghost** [reply]
Are there setup instructions for use in TrueNAS Scale? Is it possible to set up the chart without the command line?

**Reply by szaimen** [reply]
I fear I don't have a TrueNAS Scale installation available and also don't want to create one. 

So I would suggest to ask in a TrueNAS forum if there is any other way to import a custom helm chart into TruNAS scale.

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale
